src/test/mir-opt/separate_const_switch.identity.ConstProp.diff

```diff
@@ -34,107 +34,59 @@
           scope 4 {
           }
       }
-      scope 5 (inlined <Result<i32, i32> as Try>::branch) { // at $DIR/separate_const_switch.rs:29:8: 29:10
-          debug self => _4;                // in scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          let mut _10: isize;              // in scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          let _11: i32;                    // in scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          let mut _12: i32;                // in scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          let _13: i32;                    // in scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          let mut _14: std::result::Result<std::convert::Infallible, i32>; // in scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          let mut _15: i32;                // in scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          scope 6 {
-              debug v => _11;              // in scope 6 at $SRC_DIR/core/src/result.rs:LL:COL
-          }
-          scope 7 {
-              debug e => _13;              // in scope 7 at $SRC_DIR/core/src/result.rs:LL:COL
-          }
-      }
   
       bb0: {
           StorageLive(_2);                 // scope 0 at $DIR/separate_const_switch.rs:29:8: 29:10
           StorageLive(_3);                 // scope 0 at $DIR/separate_const_switch.rs:29:8: 29:10
           StorageLive(_4);                 // scope 0 at $DIR/separate_const_switch.rs:29:8: 29:9
           _4 = _1;                         // scope 0 at $DIR/separate_const_switch.rs:29:8: 29:9
-          StorageLive(_10);                // scope 0 at $DIR/separate_const_switch.rs:29:8: 29:10
-          _10 = discriminant(_4);          // scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
-          switchInt(move _10) -> [0_isize: bb5, 1_isize: bb3, otherwise: bb4]; // scope 5 at $SRC_DIR/core/src/result.rs:LL:COL
+          _3 = <Result<i32, i32> as Try>::branch(move _4) -> bb1; // scope 0 at $DIR/separate_const_switch.rs:29:8: 29:10
+                                           // mir::Constant
+                                           // + span: $DIR/separate_const_switch.rs:29:8: 29:10
+                                           // + literal: Const { ty: fn(Result<i32, i32>) -> ControlFlow<<Result<i32, i32> as Try>::Residual, <Result<i32, i32> as Try>::Output> {<Result<i32, i32> as Try>::branch}, val: Value(Scalar(<ZST>)) }
```

Comment:
Which MIR pass is causing this?

Reply:
The change causes `branch` to hit the inlining threshold and so `separate_const_switch` can't fire anymore. I'm not sure if it makes sense to give `Deinit` reduced inlining cost

Reply:
Hmm... yea, if it's that simple, just make `Deinit` not have an inlining cost I guess?

Reply:
Added a commit